packages/app/lib/login-promise/index.ts

```diff
@@ -0,0 +1,29 @@
+import { useAuth } from "app/hooks/auth/use-auth";
+import { useStableCallback } from "app/hooks/use-stable-callback";
+import { useNavigateToLogin } from "app/navigation/use-navigate-to";
+
+export let loginPromiseCallbacks = {
+  resolve: null as ((v: unknown) => void) | null,
+  reject: null as ((v: unknown) => void) | null,
+};
+
+export const useLogInPromise = () => {
+  const { accessToken } = useAuth();
```

Comment:
we use useAuth instead of useUser because of some context order issues. UserProvider is down in the hierarchy. I don't like context due to these reasons 😤 . This hook is used in useMyInfo. which I think is above the user provider at some place. It's a bit weird right now. Will spend some time cleaning it up.